src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexNode.cs

```diff
@@ -286,20 +286,20 @@ private RegexNode ReduceSet()
         {
             // Extract empty-set, one and not-one case as special
 
-            if (RegexCharClass.IsEmpty(Str))
+            if (RegexCharClass.IsEmpty(Str!))
```

Comment:
Rather than using all this `!` can we instead use a `Debug.Assert(Str != null)` at the beginning of this method? Is it actually possible that Str == null in this scope?

Reply:
`Str` can be null in general, but it looks like it has to be non-null for that particular method (RegexClass.IsEmpty would throw NRE otherwise). I'll put an assert at the top of the body.

Reply:
Done.